### PR TITLE
Fix causal-lm ONNX axis names

### PR DESCRIPTION
**Pull request by fxmarty**
As reported in: https://github.com/microsoft/onnxruntime/issues/14865#issuecomment-1460668587

### CODE REVIEW

File: optimum/exporters/onnx/base.py [hunk 1]
```diff
@@ -652,10 +652,15 @@ def add_past_key_values(self, inputs_or_outputs: Dict[str, Dict[int, str]], dire
         if direction not in ["inputs", "outputs"]:
             raise ValueError(f'direction must either be "inputs" or "outputs", but {direction} was given')
 
+        if direction == "inputs":
+            decoder_sequence_name = "past_sequence_length"
+        else:
+            decoder_sequence_name = "past_sequence_length + 1"
+
         name = "past_key_values" if direction == "inputs" else "present"
```
Comment (regisss) [inline optimum/exporters/onnx/base.py:660]: Nit: the variable `name` could be instantiated in the new conditional block right above.

File: optimum/exporters/onnx/base.py [hunk 2]
```diff
@@ -779,12 +784,15 @@ def add_past_key_values(self, inputs_or_outputs: Dict[str, Dict[int, str]], dire
 
         name = "past_key_values" if direction == "inputs" else "present"
```
Comment (regisss) [inline optimum/exporters/onnx/base.py:785]: Same here

File: optimum/exporters/onnx/model_configs.py [hunk 3]
```diff
@@ -231,15 +231,20 @@ def add_past_key_values(self, inputs_or_outputs: Dict[str, Dict[int, str]], dire
         if direction not in ["inputs", "outputs"]:
             raise ValueError(f'direction must either be "inputs" or "outputs", but {direction} was given')
 
+        if direction == "inputs":
+            decoder_sequence_name = "past_sequence_length"
+        else:
+            decoder_sequence_name = "past_sequence_length + 1"
+
         name = "past_key_values" if direction == "inputs" else "present"
```
Comment (regisss) [inline optimum/exporters/onnx/model_configs.py:239]: Same

**Review by regisss — APPROVED** [review]
LGTM!

**Review by michaelbenayoun — APPROVED** [review]
LGTM

File: optimum/exporters/onnx/base.py (hunk 1 above)
Comment (fxmarty) [reply optimum/exporters/onnx/base.py:660]: Thank you!

**Pull request merged**